### PR TITLE
BE_0001: created the master page

### PR DESCRIPTION
**Pull request by danhtrancong**

### CODE REVIEW

File: bookstore/src/main/webapp/WEB-INF/tiles.xml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      "http://tiles.apache.org/dtds/tiles-config_3_0.dtd">
+<tiles-definitions>
+
+	<!-- Base Define -->
+	<definition name="base.definition" template="/WEB-INF/layouts/classic.jsp">
+		<put-attribute name="title" value="" />
+		<put-attribute name="header" value="/WEB-INF/basefragments/header.jsp" />
+		<put-attribute name="menu" value="/WEB-INF/basefragments/menu.jsp" />
+		<put-attribute name="body" value="" />
+		<put-attribute name="footer" value="/WEB-INF/basefragments/footer.jsp" />
```
Comment (caominhnhut) [inline bookstore/src/main/webapp/WEB-INF/tiles.xml:13]: change **basefragments** to **template**

File: bookstore/src/main/webapp/WEB-INF/tiles.xml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+	<!-- Home Page -->
+	<definition name="homePage" extends="base.definition">
+		<put-attribute name="title" value="Home Page" />
+		<put-attribute name="body" value="/WEB-INF/bodyfragments/home.jsp" />
+	</definition>
+
+	<!-- ContactUs Page -->
+	<definition name="contactusPage" extends="base.definition">
+		<put-attribute name="title" value="Contact Us" />
+		<put-attribute name="body" value="/WEB-INF/bodyfragments/contactus.jsp" />
```
Comment (caominhnhut) [inline bookstore/src/main/webapp/WEB-INF/tiles.xml]: Change **bodyfragments** to **views**

File: bookstore/src/main/webapp/WEB-INF/tiles.xml (hunk 1 above)
Comment (caominhnhut) [inline bookstore/src/main/webapp/WEB-INF/tiles.xml]: Change **basefragments** to **templates**

**Review by caominhnhut — CHANGES_REQUESTED** [review]
Please fix

**Comment by caominhnhut** [comment]
Waste

**Comment by caominhnhut** [comment]
Continue to review

File: bookstore/src/main/webapp/WEB-INF/tiles.xml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+	<!-- ContactUs Page -->
+	<definition name="contactusPage" extends="bookstore.definition">
+		<put-attribute name="title" value="Liên Hệ" />
+		<put-attribute name="body" value="/WEB-INF/view/contactus.jsp" />
+	</definition>
+	
+	<!-- Login Page -->
+	<definition name="loginPage" extends="bookstore.definition">
+		<put-attribute name="title" value="login" />
```
Comment (caominhnhut) [inline bookstore/src/main/webapp/WEB-INF/tiles.xml]: The title should be changed from "login" to "User Login"
@ronaldolima99

**Review by caominhnhut — CHANGES_REQUESTED** [review]

File: bookstore/src/main/java/com/bookstore/controller/BookStoreController.java [hunk 4]
```diff
@@ -1,17 +1,30 @@
...
+		
+		model.addAttribute("address", "Vietnam");
+		model.addAttribute("phone", "09888888");
+		model.addAttribute("email", "test@gmail.com");
+		
+		return "contactusPage";
+	}
+	@RequestMapping(value = { "/register" }, method = RequestMethod.GET)
+	public String registerPage(Model model) {
+		return "registerPage";
```
Comment (caominhnhut) [inline bookstore/src/main/java/com/bookstore/controller/BookStoreController.java:28]: @danhtrancong 

1. Change method name to showRegisterView
2. Move the API to LoginController
3. Add a space between line 25, 26

File: bookstore/src/main/java/com/bookstore/controller/LoginController.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+@Controller
+@RequestMapping("/")
+public class LoginController {
+
```
Comment (caominhnhut) [inline bookstore/src/main/java/com/bookstore/controller/LoginController.java]: @ronaldolima99 
LoginController to AuthController

File: bookstore/src/main/webapp/WEB-INF/layouts/classic.jsp [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+<html>
+<head>
+	<title><tiles:getAsString name="title" /></title>
+
+	<link href="<c:url value="/css/bootstrap-4.4.1/bootstrap.min.css" />" rel="stylesheet">
+	<link href="<c:url value="/css/bootstrap-4.4.1/bootstrap-glyphicons.css" />" rel="stylesheet">
+	<link href="<c:url value="/css/jquery-ui-1.12.1.css" />" rel="stylesheet">
+	<link href="<c:url value="https://fonts.googleapis.com/css?family=Muli:300,400,500,600,700,800,900&display=swap" />" rel="stylesheet">
+    <link href="<c:url value="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" />" rel="stylesheet">
```
Comment (caominhnhut) [inline bookstore/src/main/webapp/WEB-INF/layouts/classic.jsp:14]: @danhtrancong 
Should not use the online css.
Download the css https://fonts.googleapis.com/css and "https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css use as local file.

File: bookstore/src/main/webapp/WEB-INF/template/footer.jsp [hunk 7]
```diff
@@ -0,0 +1,6 @@
+<div
+  style="background: #E0E0E0; text-align: center; padding: 5px; margin-top: 10px;">
+
```
Comment (caominhnhut) [inline bookstore/src/main/webapp/WEB-INF/template/footer.jsp]: @ronaldolima99 
Move the css to class. Not use the css in htlm directly

File: bookstore/src/main/webapp/WEB-INF/template/header.jsp [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+					<li><a href="#"><i class="fa fa-pinterest"
+							aria-hidden="true" style="color: black"></i></a></li>
+					<li><a href="#"><i class="fa fa-linkedin"
+							style="color: black"></i></a></li>
+				</ul>
+			</div>
+		</div>
+
+		<div class="col-md-2 border1-right">
+			<div class="lang">
```
Comment (caominhnhut) [inline bookstore/src/main/webapp/WEB-INF/template/header.jsp:30]: @danhtrancong 
Pickup a new name for this one: border1-right

File: bookstore/src/main/webapp/WEB-INF/template/header.jsp [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+				<button class="btn btn-secondary dropdown-toggle" type="button"
+					id="dropdownMenuButton" data-toggle="dropdown" aria-haspopup="true"
```
Comment (caominhnhut) [inline bookstore/src/main/webapp/WEB-INF/template/header.jsp:32]: @danhtrancong 
1. Remove type="button"
2. Remove id
3. Remove for the order files

File: bookstore/src/main/webapp/WEB-INF/template/header.jsp [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+								class="dropdown-item" href="#">Văn Học</a>
+						</div>
+					</div>
+				</li>
+
+				<li class="vertical"></li>
+
+				<li class="input-search"><input type="text"
+					placeholder="Tìm kiếm sản phẩm mong muốn ?" /></li>
+				<li>
```
Comment (caominhnhut) [inline bookstore/src/main/webapp/WEB-INF/template/header.jsp:72]: @ronaldolima99 
Tìm kiếm sản phẩm mong muốn ? => Search here

File: bookstore/src/main/webapp/WEB-INF/tiles.xml [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+
+	<!-- ContactUs Page -->
+	<definition name="contactusPage" extends="bookstore.definition">
+		<put-attribute name="title" value="Contact us" />
+		<put-attribute name="body" value="/WEB-INF/view/contactus.jsp" />
+	</definition>
+
+	<!-- Register Page -->
+	<definition name="registerPage" extends="bookstore.definition">
+		<put-attribute name="title" value="Register" />
```
Comment (caominhnhut) [inline bookstore/src/main/webapp/WEB-INF/tiles.xml:30]: @danhtrancong 
Register a new account

File: bookstore/src/main/webapp/static/css/style.css [hunk 12]
```diff
@@ -0,0 +1,281 @@
...
+	background-color: white;
+	padding-top: 10px;
+}
+
+.header .contact-information .login-top {
+	font-size: 15px;
+	color: #252525;
+	float: right;
+}
+.social-network ul li {
```
Comment (caominhnhut) [inline bookstore/src/main/webapp/static/css/style.css:51]: @danhtrancong 

1. Add a space between line 51, 52 and check for the other lines
2. What is parent class of .social-network

File: bookstore/src/main/webapp/static/css/style.css [hunk 13]
```diff
@@ -0,0 +1,281 @@
...
+
+.product-search .search-content {
+	height: 50px;
+	margin-top: 30px;
+	border: 1px solid #dddddd;
+	padding: 0px;
+}
+
+.search-content input {
+	height: 94.30%;
```
Comment (caominhnhut) [inline bookstore/src/main/webapp/static/css/style.css:86]: @ronaldolima99 
What is the parent class of .search-content
Should group the classes of .search-content together

File: bookstore/src/main/webapp/static/css/style.css [hunk 14]
```diff
@@ -0,0 +1,281 @@
...
+.header .contact-information .lang .lang-menu .dropdown-menu-button {
+    color: black;
+    background-color: white;
+    padding: 10.25px;
+    border: 0px;
+    font-size: 17px;
+    padding-right: 18.25px;
+}
+.dropdown-menu-button {
+    color: black;
```
Comment (caominhnhut) [inline bookstore/src/main/webapp/static/css/style.css:152]: @danhtrancong 
Where is parrent class ?

File: bookstore/src/main/webapp/static/css/style.css [hunk 15]
```diff
@@ -0,0 +1,281 @@
...
+    background-color: white;
+    padding: 10.25px;
+    border: 0px;
+    font-size: 17px;
+    padding-right: 18.25px;
+}
+.dropdown-item {
+    display: block;
```
Comment (caominhnhut) [inline bookstore/src/main/webapp/static/css/style.css:160]: @danhtrancong 
Where is parrent class ?

File: bookstore/src/main/webapp/static/css/style.css [hunk 16]
```diff
@@ -0,0 +1,281 @@
...
+    color: #212529;
+    text-align: inherit;
+    white-space: nowrap;
+    background-color: transparent;
+    border: 0;
+    font-size: 14px;
+}
+
+.navigation {
+	background-color: #252525;
```
Comment (caominhnhut) [inline bookstore/src/main/webapp/static/css/style.css:174]: @ngtang 
Where is parent class ? check for the orders

File: bookstore/src/main/webapp/static/css/style.css [hunk 17]
```diff
@@ -0,0 +1,281 @@
...
+	margin: 6px 0;
+}
+
+.body {
+	background-color: #0000FF;
+	height: 35px;
+}
+
+.login {
+	margin: auto;
```
Comment (caominhnhut) [inline bookstore/src/main/webapp/static/css/style.css:202]: @ronaldolima99 
Where is parent class ? check for the orders

File: bookstore/src/main/webapp/static/css/style.css [hunk 18]
```diff
@@ -0,0 +1,281 @@
...
+	background: #e7ab3c;
+	border: 1px solid #e7ab3c;
+	color: white;
+	outline: none;
+	margin-top: 15px;
+}
+
+/*  Register Form */
+.register-form {
+    margin: auto;
```
Comment (caominhnhut) [inline bookstore/src/main/webapp/static/css/style.css:243]: @danhtrancong 
Where is parent class ? check for the orders

**Review by caominhnhut — CHANGES_REQUESTED** [review]
Please help to fix my comments first

File: bookstore/src/main/webapp/static/css/style.css (hunk 18 above)
Comment (danhtrancong) [reply bookstore/src/main/webapp/static/css/style.css:243]: This is parent class

File: bookstore/src/main/webapp/WEB-INF/tiles.xml (hunk 3 above)
Comment (ronaldolima99) [reply bookstore/src/main/webapp/WEB-INF/tiles.xml]: done

File: bookstore/src/main/java/com/bookstore/controller/LoginController.java (hunk 5 above)
Comment (ronaldolima99) [reply bookstore/src/main/java/com/bookstore/controller/LoginController.java]: done

File: bookstore/src/main/webapp/WEB-INF/template/footer.jsp (hunk 7 above)
Comment (ronaldolima99) [reply bookstore/src/main/webapp/WEB-INF/template/footer.jsp]: done

File: bookstore/src/main/java/com/bookstore/controller/BookStoreController.java (hunk 4 above)
Comment (danhtrancong) [reply bookstore/src/main/java/com/bookstore/controller/BookStoreController.java:28]: Done

File: bookstore/src/main/webapp/WEB-INF/template/header.jsp (hunk 8 above)
Comment (danhtrancong) [reply bookstore/src/main/webapp/WEB-INF/template/header.jsp:30]: I'm going to rename: <div class="col-md-2 header-top">

**Pull request merged**